lib/src/authorization_code_grant.dart

```diff
@@ -87,18 +87,18 @@ class AuthorizationCodeGrant {
   final String _delimiter;
 
   /// The HTTP client used to make HTTP requests.
-  http.Client _httpClient;
+  http.Client? _httpClient;
 
   /// The URL to which the resource owner will be redirected after they
   /// authorize this client with the authorization server.
-  Uri _redirectEndpoint;
+  late final Uri _redirectEndpoint;
 
   /// The scopes that the client is requesting access to.
-  List<String> _scopes;
+  List<String>? _scopes;
```

Comment:
I think this should be non-nullable and set to `const []` in the constructor by default

Reply:
Or maybe `late` since it is assigned later in `getAuthorizationUrl`. What do you think?